src/python/pants/backend/python/macros/poetry_requirements_test.py

```diff
@@ -107,35 +107,43 @@ def test_handle_str(test, exp) -> None:
     assert parse_str_version(test, proj_name="foo", file_path="", extras_str="") == f"foo {exp}"
 
 
+def assert_equal_requirements(actual: str | None, expected: str) -> None:
+    assert actual is not None
+    assert PipRequirement.parse(expected) == PipRequirement.parse(actual)
```

Comment:
This is the new test. Previously the tests were ~vacuous in that the outputs were not tested against reality, which is packaging in this case (which `PipRequirement` is our wrapper around).